feat(starknet_integration_tests): refactor e2e flow test

yair-starkware opened the pull request:
Check success on aggregated txs instead of per height

yair-starkware commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/4787?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4787** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/4787?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/4787?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#4786** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/4786?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4785** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/4785?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4784** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/4784?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

reviewable-StarkWare commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/starkware-libs/sequencer/4787)
<!-- Reviewable:end -->

alonh5 reviewed (changes requested):
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 1 unresolved discussion (waiting on @ArniStarkware and @Yael-Starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OKzoty_BA6Bssxi1TTy:b8s2biv) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
> ```XML
> 
>         tokio::time::timeout(TEST_SENARIO_TIMOUT, async {
>             for tx in expected_batched_tx_hashes {
> ```

In the `many_txs_scenario` we need to check that *only*  the expected txs were batched.


<!-- Sent from Reviewable.io -->

Yael-Starkware reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @alonh5 and @ArniStarkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OL31cU049K6Z-oOSkvi:b-16070t) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

In the `many_txs_scenario` we need to check that *only*  the expected txs were batched.
</blockquote></details>

what stops the test from producing another block with the remaining txs?
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 109 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL-lTGa5xukV37Vynfe:-OL-lTGa5xukV37Vynff:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L109)):*
> ```XML
>                     }
>                     tokio::time::sleep(Duration::from_millis(2000)).await;
>                 }
> ```





_Suggestion:_
```XML
            while !expected_batched_tx_hashes.is_empty() {
                let batched_txs =
                    mock_running_system.aggregated_txs.lock().await.get_all_txs_so_far();

                expected_batched_tx_hashes.retain(|tx| !batched_txs.contains(tx));

                tokio::time::sleep(Duration::from_millis(2000)).await;
            }
```


<!-- Sent from Reviewable.io -->

Yael-Starkware reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 59 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL31nJ79UPsnqamu4DZ:-OL31nJ79UPsnqamu4D_:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L59)):*
> ```XML
>     configure_tracing().await;
> 
>     const TEST_SENARIO_TIMOUT: std::time::Duration = std::time::Duration::from_secs(50);
> ```





_Suggestion:_
```XML
TEST_SCENARIO_TIMOUT
```


<!-- Sent from Reviewable.io -->

Yael-Starkware reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 109 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL32zFs7dKv2KbnsXDu:-OL32zFs7dKv2KbnsXDv:b45h) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L109)):*
> ```XML
>                     }
>                     tokio::time::sleep(Duration::from_millis(2000)).await;
>                 }
> ```





_Suggestion:_
```XML
            loop {
                println!(
                    "Waiting for txs {} to be included in a block.",
                    expected_batched_tx_hashes
                );

                let batched_txs =
                    mock_running_system.aggregated_txs.lock().await.get_all_txs_so_far();
                expected_batched_tx_hashes.retain(|tx| !batched_txs.contains(tx));
                if expected_batched_tx_hashes.is_empty() {
                    break;
                }

                tokio::time::sleep(Duration::from_millis(2000)).await;
            }
```
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 109 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL-lTGa5xukV37Vynfe:-OL33bknAuhO3bbG2Kn8:b-y523pj) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L109)):*
> ```XML
>                     }
>                     tokio::time::sleep(Duration::from_millis(2000)).await;
>                 }
> ```

retracting this one and suggesting a refined version in the next comment.


<!-- Sent from Reviewable.io -->

alonh5 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @ArniStarkware, @Yael-Starkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OL4S8qZCANdjJh7fCy3:bfk7ip7) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

what stops the test from producing another block with the remaining txs?
</blockquote></details>

Nothing, we should test that at some point only the expected txs where batched. Or can you think of another way to verify the block was closed on size and not time?


<!-- Sent from Reviewable.io -->

yair-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @Yael-Starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLTNe2i4uSczj6rlIsZ:b4laab3) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Nothing, we should test that at some point only the expected txs where batched. Or can you think of another way to verify the block was closed on size and not time?
</blockquote></details>

How does this verify that it was closed on size and not on time?
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 109 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL32zFs7dKv2KbnsXDu:-OLTgfvK-R8pqeulaEgo:b-ey5ygw) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L109)):*
> ```XML
>                     }
>                     tokio::time::sleep(Duration::from_millis(2000)).await;
>                 }
> ```

Why is one better than the other?


<!-- Sent from Reviewable.io -->

alonh5 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 1 of 4 files reviewed, 3 unresolved discussions (waiting on @ArniStarkware, @Yael-Starkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLTqiXs0rEX7g-u8LtV:brh8doq) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

How does this verify that it was closed on size and not on time?
</blockquote></details>

Actually nothing, you're right. Maybe we should just decrease the max gas amount even more and send a lot of txs.
Also maybe we should add a metric for batches closed on time/size for the dashboard, and then we could use it here. (in a different PR). WDYT?


<!-- Sent from Reviewable.io -->

yair-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 1 of 4 files reviewed, 3 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @Yael-Starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLU-Iyq6gsYHFPcYgJW:b-vvlscn) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Actually nothing, you're right. Maybe we should just decrease the max gas amount even more and send a lot of txs.
Also maybe we should add a metric for batches closed on time/size for the dashboard, and then we could use it here. (in a different PR). WDYT?
</blockquote></details>

I can add the metric, but we can't use metrics in the flow test because sequencers override each other
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 59 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL31nJ79UPsnqamu4DZ:-OLU-DZWBCRbYeXLz8ec:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L59)):*
> ```XML
>     configure_tracing().await;
> 
>     const TEST_SENARIO_TIMOUT: std::time::Duration = std::time::Duration::from_secs(50);
> ```

Done.


<!-- Sent from Reviewable.io -->

alonh5 reviewed (changes requested):
> Reviewed 1 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @ArniStarkware, @Yael-Starkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLU4qcWD0d4FKfHxs_D:b1hrglt) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

I can add the metric, but we can't use metrics in the flow test because sequencers override each other
</blockquote></details>

It might not be exact, but we can verify that it is greater than 0 (for blocks closed on size). Also this metric will only be for the proposer so after initialization to 0 from both sequencers, there should be an override, they will just increment the counter in their turn, right?


<!-- Sent from Reviewable.io -->

alonh5 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @ArniStarkware, @Yael-Starkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLU5_tKBoRhsye2u6uC:bvtu6tf) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

It might not be exact, but we can verify that it is greater than 0 (for blocks closed on size). Also this metric will only be for the proposer so after initialization to 0 from both sequencers, there should be an override, they will just increment the counter in their turn, right?
</blockquote></details>

But anyway that's for a different PR. Here can you decrease the max gas amount and perhaps send more txs?


<!-- Sent from Reviewable.io -->

Yael-Starkware reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLYQfMtA9eQvdIi-QRV:b-ijn2pa) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

But anyway that's for a different PR. Here can you decrease the max gas amount and perhaps send more txs?
</blockquote></details>

if you know hiw many transactions you expect to get in a block (till it is full), I think it can be easily verified using the aggregated_txs , since it stores a blockNumber  to txs mapping. 
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 109 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL32zFs7dKv2KbnsXDu:-OLYPMBe0-I-h-wVzH6D:b-1i0okb) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L109)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Why is one better than the other?
</blockquote></details>

it reduces the level of nesting, since there is no for loop.


<!-- Sent from Reviewable.io -->

yair-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @Yael-Starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLcTrnH5mqFNRYfdRD2:b-pl7bwh) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

if you know hiw many transactions you expect to get in a block (till it is full), I think it can be easily verified using the aggregated_txs , since it stores a blockNumber  to txs mapping.
</blockquote></details>

I think what Yael suggests is too fragile (will break with any optimization to the system) and I deleted the mapping anyways.
Will add the metric in a separate PR.
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 109 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OL32zFs7dKv2KbnsXDu:-OLcSwEG3_AJhPH_RHgr:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L109)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

it reduces the level of nesting, since there is no for loop.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

alonh5 reviewed (changes requested):
> Reviewed 2 of 2 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: all files reviewed, 4 unresolved discussions (waiting on @ArniStarkware, @Yael-Starkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 100 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OKzoty_BA6Bssxi1TTx:-OLdBc_14XZfBhnAX-Mz:b-k6rpao) ([raw file](https://github.com/starkware-libs/sequencer/blob/9c8f12c785834e9dca57fcfc6f91d519c8d8bf07/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L100)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

I think what Yael suggests is too fragile (will break with any optimization to the system) and I deleted the mapping anyways.
Will add the metric in a separate PR.
</blockquote></details>

Sounds good, thanks!
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 108 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdCn_G4C9380E5emjN:-OLdCn_H1h-3gQinWiPz:btuz7xa) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L108)):*
> ```XML
>                 let batched_txs = &mock_running_system.aggregated_txs.lock().await.tx_hashes;
>                 expected_batched_tx_hashes.retain(|tx| !batched_txs.contains(tx));
>                 if expected_batched_tx_hashes.is_empty() {
> ```

This is cool that this way you can print which txs we're still waiting for. Can you use a new `remaining_expected_batched_tx_hashes` variable so in the end you can also assert the other direction:
```
expected_batched_tx_hashes == batched_txs
```
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 162 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdDw5244NC_zKvXzq4:-OLdDw5244NC_zKvXzq5:bi64ulk) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L162)):*
> ```XML
>         // Note: The following test scenario sends 15 transactions but only 12 are included in the
>         // block. This means that the last 3 transactions could be included in the next block if
>         // one is added to the test.
> ```

This can be removed now.



_Code quote:_
```XML
        // Note: The following test scenario sends 15 transactions but only 12 are included in the
        // block. This means that the last 3 transactions could be included in the next block if
        // one is added to the test.
```
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 163 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdE197AoaFm_OTytS7:-OLdE197AoaFm_OTytS8:b-dpn9j5) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L163)):*
> ```XML
>         // block. This means that the last 3 transactions could be included in the next block if
>         // one is added to the test.
>         TestBlockScenario {
> ```

Also these aren't blocks anymore.



_Suggestion:_
```XML
TestTxScenario
```


<!-- Sent from Reviewable.io -->

ArniStarkware commented:
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 160 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLeGO9a10nCK5Z837a8:-OLeGO9a10nCK5Z837a9:b-8qm2i4) ([raw file](https://github.com/starkware-libs/sequencer/blob/652ea110d53d062cd61b1e7e29982c07700c0f93/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L160)):*
> ```XML
> 
> fn create_test_blocks_for_many_txs_scenario() -> Vec<TestBlockScenario> {
>     vec![
> ```

There is no longer a need to split it into two test cases. (other than the fact that closing a block on size must have more than 15 txs).
This test case can be squashed into the first test case. 

Non-blocking, as this may be complicated.



_Code quote:_
```XML
fn create_test_blocks_for_many_txs_scenario() -> Vec<TestBlockScenario> {
    vec![
```


<!-- Sent from Reviewable.io -->

yair-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 0 of 5 files reviewed, 4 unresolved discussions (waiting on @alonh5, @ArniStarkware, and @Yael-Starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 108 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdCn_G4C9380E5emjN:-OLdb7JdA7UVi8iBLocT:bv4f5k0) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L108)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

This is cool that this way you can print which txs we're still waiting for. Can you use a new `remaining_expected_batched_tx_hashes` variable so in the end you can also assert the other direction:
```
expected_batched_tx_hashes == batched_txs
```
</blockquote></details>

It's a problem in the too_many_txs test because the additional txs get to the accumulated txs too.
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 162 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdDw5244NC_zKvXzq4:-OLmN1vk4qgENP7kiaRf:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L162)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

This can be removed now.
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 163 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdE197AoaFm_OTytS7:-OLmNFpXCbz1OwI0m8je:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L163)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Also these aren't blocks anymore.
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 160 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLeGO9a10nCK5Z837a8:-OLmmykk2OZ5_kAVQIBG:b-entbn2) ([raw file](https://github.com/starkware-libs/sequencer/blob/652ea110d53d062cd61b1e7e29982c07700c0f93/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L160)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

There is no longer a need to split it into two test cases. (other than the fact that closing a block on size must have more than 15 txs).
This test case can be squashed into the first test case. 

Non-blocking, as this may be complicated.
</blockquote></details>

Can't because limiting the `block_max_capacity_sierra_gas` for the too_many_txs is causing the l1handler tx to fail in the batcher


<!-- Sent from Reviewable.io -->

alonh5 reviewed (changes requested):
> Reviewed 3 of 5 files at r7, 2 of 3 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: all files reviewed, 2 unresolved discussions (waiting on @ArniStarkware, @Yael-Starkware, and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 108 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdCn_G4C9380E5emjN:-OLms5WR4l6iQ0DB6Dtc:blmgdpr) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L108)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

It's a problem in the too_many_txs test because the additional txs get to the accumulated txs too.
</blockquote></details>

Like we said we aren't checking those 12 tx are in one block anyways here right?
You can decrease even more the gas amount and send more txs just to be sure the block will close on size, then you can add this assertion.
In a separate PR can we check the logs we're getting a block full log?


<!-- Sent from Reviewable.io -->

ArniStarkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: all files reviewed, 1 unresolved discussion (waiting on @Yael-Starkware and @yair-starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 160 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLeGO9a10nCK5Z837a8:-OLnvHzh4YrUomUuSLWo:bhvf5k0) ([raw file](https://github.com/starkware-libs/sequencer/blob/652ea110d53d062cd61b1e7e29982c07700c0f93/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L160)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Can't because limiting the `block_max_capacity_sierra_gas` for the too_many_txs is causing the l1handler tx to fail in the batcher
</blockquote></details>

Can you increase the number of Txs be more than 15?
Also not sure - because that case the block might be closed on time before it is closed on size.
But, as I said - out of scope.


<!-- Sent from Reviewable.io -->

yair-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787)* status: 1 of 5 files reviewed, 1 unresolved discussion (waiting on @alonh5 and @Yael-Starkware)

___
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 108 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdCn_G4C9380E5emjN:-OLnv0DL51KO7X8LKJPw:b315w4m) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L108)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Like we said we aren't checking those 12 tx are in one block anyways here right?
You can decrease even more the gas amount and send more txs just to be sure the block will close on size, then you can add this assertion.
In a separate PR can we check the logs we're getting a block full log?
</blockquote></details>

Added a todo and will try it in a separate PR


<!-- Sent from Reviewable.io -->

yair-starkware commented:
*[`crates/starknet_integration_tests/tests/end_to_end_flow_test.rs` line 108 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-OLdCn_G4C9380E5emjN:-OM1PE0T30TSj0HGxIft:b-keeg4d) ([raw file](https://github.com/starkware-libs/sequencer/blob/3711fe0b6790aa6b575be8ef89e489ed9268189c/crates/starknet_integration_tests/tests/end_to_end_flow_test.rs#L108)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Added a todo and will try it in a separate PR
</blockquote></details>

Added the check of the other direction


<!-- Sent from Reviewable.io -->

alonh5 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/4787#-:-OMLjvssBJCS9dohY04X:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Yael-Starkware)



<!-- Sent from Reviewable.io -->
